ci/tests/disklayout/disk_roles_test.py

```diff
@@ -0,0 +1,256 @@
+        # End validate input
+
+        logger.info("Starting {0} of disk roles".format(operation_type))
+        collection = {}
+
+        for key, value in config.iteritems():
+            # key = ip address
+            # value = disks
+
+            for disk_info in value['disks']:
+                roles_list = disk_info['roles']
+                disk_name = disk_info['disk_name']
+                logger.info("Fetching disk with diskname '{0}' for ip '{1}'".format(disk_name, key))
+                disk = GeneralStorageRouter().get_disk_by_ip(disk_name, key)
+                logger.info("Fetching or creating new partitions for disk '{0}'".format(disk.guid))
+                partition = GeneralDisk.partition_disk(disk)
+                new_roles_list = None
+                if len(partition.roles) > 0:
+                    original_roles = partition.roles
+                    logger.info("Found roles '{0}' on partition '{1}'".format(original_roles, partition.guid))
+                    new_roles_list = list(roles_list)
+                    for role in original_roles:
+                        if role not in new_roles_list:
+                            new_roles_list.append(role)
+                logger.info("Adding roles '{0}' to partition '{1}'".format(roles_list, partition.guid))
+                collection[partition.guid] = roles_list
+                if operation_type == "APPEND":
+                    GeneralDisk.adjust_disk_role(partition, roles_list, 'APPEND')
+                    if new_roles_list:
+                        collection[partition.guid] = new_roles_list
+                elif operation_type == "SET":
+                    GeneralDisk.adjust_disk_role(partition, roles_list, 'SET')
+        return collection
+
+    @staticmethod
+    def remove_roles_from_config(config, number_of_roles_to_remain=0):
+        """
+
+        :param config: Configuration file containing all of the required information
+        :type config: dict
+
+        :param number_of_roles_to_remain: how roles may still be defined on the partition. The first 'number_of_roles_to_remain' will remain.
+        :type number_of_roles_to_remain: int
+
+        :return: Returns a object with the partition guid and its roles
+        :type: Object
+        """
+        collection = {}
+        # Remove disk roles
+        logger.info("Starting removal of disk roles")
+        for key, value in config.iteritems():
+            for disk_info in value['disks']:
+                roles_list = disk_info['roles']
+                disk_name = disk_info['disk_name']
+                logger.info("Fetching disk with diskname '{0}' for ip '{1}'".format(disk_name, key))
+                disk = GeneralStorageRouter().get_disk_by_ip(disk_name, key)
+                logger.info("Fetching or creating new partitions for disk '{0}'".format(disk.guid))
+                partition = GeneralDisk.partition_disk(disk)
+                if not number_of_roles_to_remain == 0:
```

Comment:
I'd prefer positive logic, which is more readable:
if ... == 0:
   do x
else:
   do y